// codethecouture-server/src/data.ts
export const CATEGORIES = [
  {
    id: 1,
    name: "Climate Survival & Adaptation",
    puzzles: [
      { text: "<b>Equation:</b> Weather ÷ 30 Years = ?<br/><b>Logic:</b> I am the \"personality\" of a region, while weather is just its mood today.", answer: "CLIMATE" },
      { text: "<b>Condition:</b> (Needs > Wants) + Oxygen = Success.<br/><b>Logic:</b> The only game where \"breaking even\" is a total victory. Herbert Spences says … of the fittest.", answer: "SURVIVAL" },
      { text: "<b>Logic:</b> If Environment = X and Organism = Y, then Y must become X+1.<br/><b>Wordplay:</b> The \"Update\" button for living things... From long necks of giraffes to the streamlined bodies of fish.", answer: "ADAPTATION" }
    ],
    themes: ["The Air We Wear", "Water on the Edge", "Drowning World", "Earth on Fever", "Disappearing Seasons", "Grow Your Own Oxygen"]
  },
  {
    id: 2,
    name: "Nature, Wildlife & Ecosystems",
    puzzles: [
      { text: "<b>Equation:</b> Earth - Concrete = ?<br/><b>Logic:</b> The only factory that builds, fuels, and cleans itself for free.", answer: "NATURE" },
      { text: "<b>Equation:</b> Biology × Freedom = ?<br/><b>Logic:</b> The 5 sensed residents of a city with no streetlights, no laws, and no Wi-Fi. Nat Geo …?", answer: "WILDLIFE" },
      { text: "<b>The Web:</b> Plant + Animal + Soil × Interaction = ?<br/><b>Logic:</b> A house where every resident is also the foundation. Your EVS book chapter.", answer: "ECOSYSTEM" }
    ],
    themes: ["Nature Strikes Back", "Last Tree Standing", "Animal in Crisis", "Plastic Mermaid", "Flow of Power"]
  },
  {
    id: 3,
    name: "Waste, Pollution & Circularity",
    puzzles: [
      { text: "<b>Logic:</b> I am a human invention. In the woods, I don't exist; in the city, I'm everywhere.<br/><b>Equation:</b> Resource + Wrong Location = ?", answer: "WASTE" },
      { text: "<b>Wordplay:</b> The unwanted \"plus one\" at every industrial party.<br/><b>Equation:</b> Clean Air + Human Output = Toxin. An issue in air, water, land and noise.", answer: "POLLUTION" },
      { text: "<b>Logic:</b> Output → Input → Output. The geometry of a world without a trash can. Every exit is an entrance. Ability of O. An economic and industrial model to eliminate waste.", answer: "CIRCULARITY" }
    ],
    themes: ["Walking Landfill", "Power from Waste", "Consumerism Monster", "The Walking Billboard", "CC - Clickbait Couture"]
  },
  {
    id: 4,
    name: "Energy & Resource System",
    puzzles: [
      { text: "<b>Law:</b> I can never be created or destroyed, only \"traded\" for a different look.<br/><b>Relativity Equation:</b> Mass × Light Speed² = ?", answer: "ENERGY" },
      { text: "<b>The Vault:</b> Raw Matter + Human Need = ?<br/><b>Wordplay:</b> The \"Ingredients\" list for everything ever manufactured.", answer: "RESOURCE" },
      { text: "<b>A “Stem” Structure:</b> Rule A + Rule B = Predictable Output.<br/><b>Logic:</b> The skeleton of the digital world; without me, data has no shape. R Maan’s song.", answer: "SYSTEM" }
    ],
    themes: ["Solar Saviour", "Wind Walker", "Human Battery", "Light in the Dark", "Fire Within", "Energy Drained Society"]
  },
  {
    id: 5,
    name: "Human vs Machine Identity",
    puzzles: [
      { text: "<b>The Flaw:</b> Logic + Emotion + Mistakes = ?<br/><b>Wordplay:</b> The only computer that crashes because of a \"feeling.\" You are it. I am it. Your teammate is also it.", answer: "HUMAN" },
      { text: "<b>The Tool:</b> Metal + Code - Intuition = ?<br/><b>Logic:</b> I can count a billion stars, but I don't know why they are beautiful.", answer: "MACHINE" },
      { text: "<b>The Core:</b> Who I am - What I look like = ?<br/><b>Wordplay:</b> The \"User Name\" of your soul.", answer: "IDENTITY" }
    ],
    themes: ["Half Code, Half Human", "Bio-Hacked Human", "CC - Cyborg Confidence", "Human Becoming Data", "Emotionless Efficiency"]
  },
  {
    id: 6,
    name: "AI Control, Algorithms & Systems",
    puzzles: [
      { text: "<b>Logic:</b> A brain smarter than us, but still following our orders.<br/><b>Wordplay:</b> What does your mom say when you show her a reel of progressive parents? “Ye .. hain.”", answer: "AI" },
      { text: "<b>The Pattern:</b> Input → Decision Tree → Result.<br/><b>Wordplay:</b> The invisible chef following a recipe you never gave me. It is how your IG feed is built.", answer: "ALGORITHM" },
      { text: "<b>A “Stem” Structure:</b> Rule A + Rule B = Predictable Output.<br/><b>Logic:</b> The skeleton of the digital world; without me, data has no shape. R Maan’s song.", answer: "SYSTEM" }
    ],
    themes: ["Watched All the Time", "Controlled by the Feed", "Stuck in a Loop", "War Against AI", "Digital Puppet Master"]
  },
  {
    id: 7,
    name: "Digital & Online Behaviour",
    puzzles: [
      { text: "<b>Logic:</b> Physical World - Matter = ?<br/><b>Logic:</b> I am a world made of binary code where distance is measured in milliseconds, not miles. One thing you’ll remember when you hear ‘Analogue clocks’, P.S. they are my family members.", answer: "DIGITAL" },
      { text: "<b>The Connection:</b> Device + Signal = Everywhere.<br/><b>Wordplay:</b> A place you \"go\" to without ever leaving your chair. Your ex’s app status after they say they’re going to sleep soon tonight.", answer: "ONLINE" },
      { text: "<b>The Trail:</b> Choice × Repeat = Data Profile.<br/><b>Logic:</b> What you do when you think only a screen is watching.", answer: "BEHAVIOUR" }
    ],
    themes: ["Filtered Life", "Seen but Not Known", "Validation Machine", "Viral or Invisible", "POV Content", "Instagram Stories"]
  },
  {
    id: 8,
    name: "Internet, Content & Culture",
    puzzles: [
      { text: "<b>Equation:</b> World Wide Web + Human Behavior = ?<br/><b>Logic:</b> The \"vibe\" of the collective screen; a coffee shop with 5 billion people all shouting at once.", answer: "INTERNET" },
      { text: "<b>Equation:</b> Creativity ÷ Attention Span = ?<br/><b>Logic:</b> The \"filler\" for the digital void. If you stop scrolling, I stop existing.", answer: "CONTENT" },
      { text: "<b>Equation:</b> Ideas + Speed + Irony = ?<br/><b>Logic:</b> The invisible \"team spirit\" of the web that decides what is funny today and forgotten tomorrow.", answer: "CULTURE" }
    ],
    themes: ["Fandom Culture", "Anti-Hero Era", "Edits Culture", "Internet Memes", "Doom Scrolling", "Scrolling Life Away"]
  },
  {
    id: 9,
    name: "Economy, Work & Inequality",
    puzzles: [
      { text: "<b>The Machine:</b> Trust × Exchange = ?<br/><b>Logic:</b> The giant game of \"trading\" that keeps the world's lights on. Jiran Seyn’s recent song.", answer: "ECONOMY" },
      { text: "<b>The Swap:</b> Energy + Hours = Income.<br/><b>Wordplay:</b> What you do to afford the life you’re too busy to live. Rihanna’s old song.", answer: "WORK" },
      { text: "<b>The Gap:</b> Wealth ÷ Opportunity ≠ 1.<br/><b>Logic:</b> The distance between the top of the ladder and the person who can’t reach it. Exists as an issue in many forms: gender, caste etc.", answer: "INEQUALITY" }
    ],
    themes: ["Walking Economy Crash", "The Invisible Worker", "Rich vs Poor Divide", "Bill After Bill"]
  },
  {
    id: 10,
    name: "Society, Identity & Psychology",
    puzzles: [
      { text: "<b>The Social Collective:</b> Individual × 8 Billion = ?<br/><b>Logic:</b> The imaginary \"team\" that everyone agrees to play for.", answer: "SOCIETY" },
      { text: "<b>The Mirror:</b> Memory + Action = ?<br/><b>Wordplay:</b> The only thing you own that you can never see without help. You always carry it as a card.", answer: "IDENTITY" },
      { text: "<b>The Blueprint:</b> Brain + Experience = Why.<br/><b>Logic:</b> The science of thinking and figuring out why you just did what you did.", answer: "PSYCHOLOGY" }
    ],
    themes: ["Crowd, Not Individual", "Thought ≠ Reality", "Beyond Labels", "Overthinking Society", "People Pleaser", "Brain Overload", "Main Character Syndrome"]
  }
];

export const MAX_SLOTS_PER_THEME = 3;
